src/lib/BasicSide.tsx

```diff
@@ -101,195 +53,40 @@ function BasicSide(props: any) {
   const navigate = useNavigate()
   const location = useLocation()
 
-  const { frame } = spec
-
-  // spec schema definition with Gubu
-  const shape = Gubu({
-    side: {
-      logo: { img: "" },
-      section: [
-        { name: "", kind: "", view: {}, button: { icon: String, text: String } },
-      ]
-    },
-    view: {}
-  })
-
-  // spec schema validation with Gubu
-  shape(spec)
-
-  const part = spec.side
-
-  const viewmap = spec.view
-  const viewdefs = Object.entries(viewmap)
-    .map((entry: any) => (entry[1].name = entry[0], entry[1]))
+  BasicSideSpecShape(spec)
```

Comment:
lets make it clear that spec is possibly mutated by the addition of default values
```
spec = BasicSideSpecShape(spec)
```

Reply:
Got it! So we might end up using Gubu for more than validation.

Reply:
I ended up doing: 

```
  const specWithDefaults = BasicSideSpecShape(spec)
```

As TS was complaining `spec` is a constant.

Reply:
Fixed ✅